src/components/ProgressBar/index.tsx

```diff
@@ -110,29 +179,14 @@ const ProgressBar: FunctionComponent<ProgressBarProps> = ({
                 </div>
             );
         }
-        return (
-            <Grid container spacing={3}>
-                <Grid item xs={value && displayValue ? 11 : 12}>
-                    <LinearProgress
-                        variant="determinate"
-                        value={value || 100}
-                        classes={{
-                            colorPrimary: classes.colorPrimary,
-                            barColorPrimary: classes.barColorPrimary,
-                        }}
-                    />
-                </Grid>
-                {displayValue && value && (
-                    <Grid item xs={1}>
-                        <Text>{value}%</Text>
-                    </Grid>
-                )}
-            </Grid>
-        );
+
+        const ProgressBarComponent = variant === 'linear' ? LinearProgressComponent : CircularProgressWithLabel;
```

Comment:
use useCallback to avoid update on each render.

Reply:
Changed to useMemo given this is not a function.